docs/articles/immutability-in-js.md

```diff
@@ -0,0 +1,108 @@
+# Immutability in JavaScript
+
+According to (wikipédia)[https://en.wikipedia.org/wiki/Immutable_object] an *immutable object (unchangeable object) is an object whose state cannot be modified after it is created*. This rule is quite simple, if you want to modify some property of an object you have to do it on a copy. We will see a bit later what improvements and fancy features it unlocks for our developments.
+
+## EcmaScript
+
+EcmaScript provides utilities to keep our data immutable. APIs of arrays and objects have methods to create copies and even prevent instances to be updated. More recently EcmaScript brings us a brand new syntax that makes it easier to achieve it.
+
+### Object.assign
+
+We want to add a name property in our object.
+
+https://gist.github.com/frinyvonnick/a281f15288d3ab90ce5354f01df52ccf
+
+We can do it with `Object.assign` and a little trick. Basically it copies all properties of an object to another one, thus it mutates the target object. So we use a small trick, by passing an empty object at first parameter we create a new reference.
```

Comment:
`by passing an empty object at first parameter we create a new reference` => il manque de la ponctuation, non ? :smile: 
est-ce que ce ne serait pas mieux avec `So we use a small trick: passing an empty object at first parameter, which creates a new reference`